pkg/providers/amifamily/bootstrap/windows.go

```diff
@@ -0,0 +1,80 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+
+	"github.com/samber/lo"
+)
+
+type Windows struct {
+	Options
+}
+
+// nolint:gocyclo
+func (w Windows) Script() (string, error) {
+	var userData bytes.Buffer
+	userData.WriteString("<powershell>\n")
+	userData.WriteString("[string]$EKSBootstrapScriptFile = \"$env:ProgramFiles\\Amazon\\EKS\\Start-EKSBootstrap.ps1\"\n")
+	userData.WriteString(fmt.Sprintf(`& $EKSBootstrapScriptFile -EKSClusterName "%s" -APIServerEndpoint "%s"`, w.ClusterName, w.ClusterEndpoint))
+	if w.CABundle != nil {
+		userData.WriteString(fmt.Sprintf(" -Base64ClusterCA \"%s\"", *w.CABundle))
+	}
+
+	kubeletExtraArgs := strings.Join([]string{w.nodeLabelArg(), w.nodeTaintArg()}, " ")
+	if kubeletExtraArgs = strings.Trim(kubeletExtraArgs, " "); len(kubeletExtraArgs) > 0 {
+		userData.WriteString(fmt.Sprintf(` -KubeletExtraArgs "%s"`, kubeletExtraArgs))
+	}
+	if w.KubeletConfig != nil && len(w.KubeletConfig.ClusterDNS) > 0 {
+		userData.WriteString(fmt.Sprintf(` -DNSClusterIP "%s"`, w.KubeletConfig.ClusterDNS[0]))
+	}
+	userData.WriteString("\n</powershell>")
```

Comment:
In addition to above, there is a parameter `ServiceCIDR` which needs to be present in the userdata. This is required in case the user is using a custom CIDR range instead of default one.

Reference: https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-windows-ami.html#bootstrap-script-configuration-parameters

Reply:
This is reasonable, however, there's no such ServiceCIDR option in Kapenter framework. Not sure how to get this config in Kapenter

Reply:
This setting can be obtained similar to `DNSClusterIP` and is part of the response of DescribeCluster API.

@jonathan-innis can probably suggest the location from where we can plumb this value.

Reply:
@rawahars 
I think this requires changing the v1alpha5/machines.go KubeletConfiguration in the karpenter core first.

If Service CIDR is required, it's better to change the core first.